src/routes.js

```diff
@@ -35,17 +36,30 @@ router.get('/coffeegrams/new', async ctx => {
 });
 
 router.post('/coffeegrams', async ctx => {
-  var form = ctx.request.body;
+  var { image, description, type, shop } = ctx.request.body;
+
+  var coffeeParams = {
+    image,
+    description: clean(description),
+    type: clean(type),
+    shop: clean(shop)
+  }
+
+  var errors = validateImage(coffeeParams);
+
+  if (Object.keys(errors).length) {
+    return await ctx.render('new', ({ errors, coffeeParams }));
+  }
+
   await Coffeegram.create({
     userId: ctx.state.currentUser.id,
-    image: path.basename(form.image[0].path),
-    description: form['description'],
-    type: form['type'],
-    shop: form['shop']
+    image: path.basename(image[0].path),
+    description,
```

Comment:
Uh oh, these won't be cleaned anymore and you've opened yourself to security issues. I think you want to do `description: coffeeParams.description`?